_data/rules.yml

```diff
@@ -295,6 +295,50 @@ services:
   - name: Kubernetes
     exporters:
       - rules:
+          - name: KubeAPIServerDown
+            description: "Kubernetes API has disappeared from Prometheus target discovery."
+            query: 'absent(up{job="kubernetes-apiservers"}) == 1'
+            severity: critical
+          - name: KubeAPIServerLatency
+            description: "Kubernetes API server has a 99th percentile latency of {{ $value }} seconds for {{ $labels.verb }} {{ $labels.resource }}."
+            query: 'histogram_quantile(0.99, sum(apiserver_request_latencies_bucket{verb!~"CONNECT|WATCHLIST|WATCH|PROXY"}) WITHOUT (instance, resource)) / 1e+06 > 1'
+            severity: warning
+          - name: KubeAPIServerLatencyHigh
+            description: "Kubernetes API server has a 99th percentile latency of {{ $value }} seconds for {{ $labels.verb }} {{ $labels.resource }}."
+            query: 'histogram_quantile(0.99, sum(apiserver_request_latencies_bucket{verb!~"CONNECT|WATCHLIST|WATCH|PROXY"}) WITHOUT (instance, resource)) / 1e+06 > 4'
+            severity: critical
+          - name: KubeClientErrors
+            description: "Kubernetes API server client '{{ $labels.job }}/{{ $labels.instance }}' is experiencing {{ $value | humanize }}% errors.'"
+            query: '(sum(rate(rest_client_requests_total{job=~"kubernetes-.*",code=~"5.."}[5m])) by (instance, job) / sum(rate(rest_client_requests_total[5m])) by (instance, job)) * 100 > 1'
+            severity: warning
+          - name: KubeControllerManagerDown
+            description: "Kubernetes controller manager has disappeared from Prometheus target discovery."
+            query: 'absent(up{k8s-app="kubernetes-controller-manager"}) == 1'
+            severity: critical
+          - name: KubeNodeNotReady
+            description: "{{ $labels.node }} has been unready for more than an hour."
+            query: 'kube_node_status_condition{condition="Ready",status="true"} == 0'
+            severity: warning
+          - name: KubeSchedulerDown
+            description: "Kubernetes scheduler has disappeared from Prometheus target discovery."
+            query: 'absent(up{k8s-app="kubernetes-scheduler"}) == 1'
```

Comment:
Is it possible to have something more generic here ?
Whatever the value of `k8s-app` label is ?

Same remark for KubeStateMetricsDown and KubeControllerManagerDown ?